packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -2601,6 +2602,19 @@ class StackViewport extends Viewport implements IStackViewport, IImagesLoader {
       }, 40);
     }
 
+    const imageIdIndex = this.getCurrentImageIdIndex();
+
+    if (
+      imageIdIndex + delta > this.getImageIds().length - 1 ||
+      imageIdIndex + delta < 0
+    ) {
+      const eventData: StackScrollOutOfBoundsEventDetail = {
+        imageIdIndex,
+        direction: delta,
+      };
+      triggerEvent(this.element, Events.STACK_SCROLL_OUT_OF_BOUNDS, eventData);
+    }
```

Comment:
I'm not sure if this makes sense, it is talking about a tool event inside the viewport, i think this should not happen here, but rather in the scroll() utilitiy